File: problem/Maximum_Erasure_Value/maximumUniqueSubarray.c
int maximumUniqueSubarray(int *nums, int numsSize)
{
    int seen[10001] = {0};
    int left = 0, currentSum = 0, maxSum = 0;
    for (int right = 0; right < numsSize; right++)
    {
        while (seen[nums[right]])
        {
            seen[nums[left]] = 0;
            currentSum -= nums[left];
            left++;
        }
        seen[nums[right]] = 1;
        currentSum += nums[right];
        if (currentSum > maxSum)
            maxSum = currentSum;
    }
    return maxSum;
}

File: problem/Maximum_Erasure_Value/maximumUniqueSubarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maximumUniqueSubarray = function(nums) {
  const seen = new Set();
  let left = 0, currentSum = 0, maxSum = 0;

  for (let right = 0; right < nums.length; right++) {
    while (seen.has(nums[right])) {
      seen.delete(nums[left]);
      currentSum -= nums[left];
      left++;
    }
    seen.add(nums[right]);
    currentSum += nums[right];
    maxSum = Math.max(maxSum, currentSum);
  }

  return maxSum;
};